cmd/ingest/batch.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	ctx509 "github.com/google/certificate-transparency-go/x509"
+	"github.com/netsec-ethz/fpki/pkg/common"
+	"github.com/netsec-ethz/fpki/pkg/db"
+	mcommon "github.com/netsec-ethz/fpki/pkg/mapserver/common"
+	"github.com/netsec-ethz/fpki/pkg/mapserver/updater"
+)
+
+const BatchSize = 10000
+
+type Batch struct {
+	Certs  []*ctx509.Certificate
+	Chains [][]*ctx509.Certificate
+	cns    []*string
+}
+
+func NewBatch() *Batch {
+	return &Batch{
+		Certs:  make([]*ctx509.Certificate, 0, BatchSize),
+		Chains: make([][]*ctx509.Certificate, 0, BatchSize),
+		cns:    make([]*string, 0, BatchSize),
+	}
+}
+
+// AddData pushed the cert data into the batch.
+func (b *Batch) AddData(d *CertData) {
+	b.Certs = append(b.Certs, d.Cert)
+	b.Chains = append(b.Chains, d.CertChain)
+	b.cns = append(b.cns, &d.Cert.Subject.CommonName)
```

Comment:
Here we are ignoring SAN entries. You could use the [`extractCertDomains` ](https://github.com/netsec-ethz/fpki/blob/juagargi/ingest_tool/pkg/mapserver/updater/tools.go#L12) function to extract CN + SAN.